tasks/push-sbom-to-pyxis/push-sbom-to-pyxis.yaml

```diff
@@ -48,31 +41,28 @@ spec:
         #!/usr/bin/env sh
         set -eux
 
-        SNAPSHOT_SPEC_FILE="$(workspaces.data.path)/$(params.snapshotPath)"
-        if [ ! -f "${SNAPSHOT_SPEC_FILE}" ] ; then
-            echo "No valid snapshot file was provided."
-            exit 1
-        fi
-
-        IMAGEURLS=($(cat "${SNAPSHOT_SPEC_FILE}" | jq -r '.components[].containerImage'))
-        IMAGEIDS=($(params.containerImageIDs))
-
-        if [ ${#IMAGEURLS[@]} != ${#IMAGEIDS[@]} ]; then
-          echo "ERROR: Lengths of imageURLs and containerImageIDs do not match"
-          exit 1
-        fi
+        PYXIS_JSON_PATH=$(workspaces.data.path)/pyxis.json
 
         mkdir /workdir/sboms
         cd /workdir/sboms
 
-        for i in ${!IMAGEURLS[@]}; do
-          echo "Fetching sbom for image: ${IMAGEURLS[$i]}"
-          cosign download sbom --output-file "${IMAGEIDS[$i]}.json" "${IMAGEURLS[$i]}"
+        IMAGEURL_LENGTH=($(jq -r '.components | length' $PYXIS_JSON_PATH))
```

Comment:
Is there any benefit in traversing the components like this instead of a for each like we do here https://github.com/redhat-appstudio/release-service-catalog/blob/development/tasks/push-snapshot/push-snapshot.yaml#L94?

Reply:
I see this is a safer way to get the values from the JSON without need to trust that the string is not carrying any char that would eventually cause the shell to misread it. It is common you need to apply some bash modifiers to force `read` to get the strings the right way.

Maybe for this us unnecessary for this specific JSON, but as generally speaking it could be, I prefer to keep the 
 - maybe good - habbit.